Plugin/TriggerViewSearchResultDataLayerEvent.php

```diff
@@ -19,10 +20,11 @@ public function __construct(
         $this->customerSessionDataProvider = $customerSessionDataProvider;
     }
 
-    public function afterExecute(Index $subject): void
+    public function afterExecute(Index $subject, AbstractResult|null $result): ?AbstractResult
```

Comment:
It must complies with the ActionInterface signature:
`\Magento\Framework\Controller\ResultInterface|\Magento\Framework\App\ResponseInterface`

Reply:
It seems to be a bad idea because  
`\Magento\CatalogSearch\Controller\Result\Index::execute `

As no such signature and allow tu return anything

`/**
     * Display search result
     *
     * @return void
     *
     * @throws \Magento\Framework\Exception\LocalizedException
     */
    public function execute()
    {`

Reply:
So i put a "mixed" type [on this commit](https://github.com/yireo/Yireo_GoogleTagManager2/pull/214/commits/9271a2b58f01e953cebc0bc4457d0493d7e187c1) in order to avoid any typing issue